### PR TITLE
Change aux key to -x-

### PR DESCRIPTION
**Pull request by sffc**
#3632

### CODE REVIEW

**Comment by sffc** [comment]
Breaks transliterator, not the compilation but one of the tests. Will debug tomorrow.

**Comment by sffc** [comment]
Note that this requires auxiliary keys to be lowercase since UTS-35 requires that the whole string is lowercase except for the region and script.

**Comment by zbraniecki** [comment]
why are you removing emojis from hello world samples?

**Comment by sffc** [comment]
> why are you removing emojis from hello world samples?

Not intentional. Probably a bug in datagen with the new aux keys. Will investigate. Thanks for noticing :)

File: provider/datagen/src/baked_exporter.rs [hunk 1]
```diff
@@ -462,10 +462,7 @@ impl BakedExporter {
                     &first_locale
                         .chars()
                         .flat_map(|ch| {
-                            if ch == AuxiliaryKeys::separator() as char {
-                                // Replace the aux key separator with double-underscore
-                                ['_'].into_iter().chain(Some('_'))
-                            } else if ch == '-' {
+                            if ch == '-' {
```
Comment (robertbastian) [inline provider/datagen/src/baked_exporter.rs:465]: the `flat_map` and `chain(None)` gymnastics can be removed

**Review by robertbastian — COMMENTED** [review]
I was hoping we could reuse a bit more of the `icu_locid::extensions::private::Private` type. We already carry the code for it through `Locale`, so you shouldn't need to implement parsing or stringifying here.

**Comment by sffc** [comment]
The API doesn't force a particular underlying structure so for now I left it the same along with pre-existing comments suggesting how it could change after we take measurements.

**Comment by robertbastian** [comment]
My comment is not primarily about performance. It's easier to maintain code if we don't have it twice. And because the API doesn't force a particular structure we can just use a `extensions::private::Private`, like we use `extensions::unicode::Keywords`.

**Review by Manishearth — DISMISSED** [review]
Much simpler, nice

**Comment by sffc** [comment]
Need to look at the test failure; probably need to regen data or something.

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
The bug causing the test failures and the disappearance of the emoji was https://github.com/unicode-org/icu4x/pull/4144/commits/3efa28e9158ba9ea2c3f153399995fec793be2e5

File: provider/datagen/src/baked_exporter.rs (hunk 1 above)
Comment (sffc) [reply provider/datagen/src/baked_exporter.rs]: Done

**Comment by sffc** [comment]
Finally green. Landable after the next approval

**Review by Manishearth — APPROVED** [review]

**Pull request merged**